pl_examples/hydra_examples/pl_template.py

```diff
@@ -0,0 +1,48 @@
+"""
+Pytorch Lightning training using Hydra for configuration
+"""
+
+import hydra
+import pl_examples.hydra_examples.user_config
+from omegaconf import DictConfig, OmegaConf
+from hydra.utils import instantiate
+
+
+from pl_examples.models.hydra_config_model import LightningTemplateModel
+from pytorch_lightning import Callback, seed_everything, Trainer
+
+
+@hydra.main(config_path="conf", config_name="config")
+def main(cfg: DictConfig):
+    """
+    Main training routine specific for this project
+    :param cfg:
+    """
+    seed_everything(cfg.model.seed)
+    # ------------------------
+    # 1 INIT LIGHTNING MODEL
+    # ------------------------
+    model = LightningTemplateModel(cfg.model, cfg.data, cfg.scheduler, cfg.opt)
+    # ------------------------
+    # 2 INIT TRAINER
+    # ------------------------
+ 
+    callbacks = [instantiate(c) for c in cfg.callbacks.callbacks_list] if  cfg.callbacks else []
+
+    trainer = Trainer(
+        **cfg.trainer,
+        logger=instantiate(cfg.logger),
+        profiler=instantiate(cfg.profiler),
+        checkpoint_callback=instantiate(cfg.checkpoint),
+        early_stop_callback=instantiate(cfg.early_stopping),
+        callbacks=callbacks,
+    )
```

Comment:
PL Team, any thoughts about simplifying this?
Is it possible to create standard config driven initialization method (or factory) for the trainer?

How do you feel about:
```python
trainer  = Trainer.from_cfg(cfg)
```

@anthonytec2 :
I think all the PL specific configs should be under the same node:
```yaml
pl:
  trainer:
     ...
  logger:
    ...
  profiler:
    ...
  ...  
```

This would allow more easier initialization of PL (without having to worry about the user adding his/her own things to the top level config:
```python
trainer  = Trainer.from_cfg(cfg.pl)
```

Reply:
Actually I like this proposal, but I feel we should change the name to `Trainer.from_hydra_config` since otherwise it may not be clear, what kind of config to expect here.

Reply:
Since we have already a Trainer.from_argparse_args what about Trainer.from_hydra for the name? Or is config meant to be more general?

Reply:
I thought the same. I agree with Justus

Reply:
The config object is just an object with some fields, there is nothing Hydra about it.